src/Attach.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2017 Sasha Goldshtein
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import java.io.File;
+
+import com.sun.tools.attach.VirtualMachine;
+import java.lang.management.ManagementFactory;
+
+public class Attach {
+    public static void main(String[] args) throws Exception {
+        if (args.length < 2) {
+            System.out.println("USAGE: <agent-path> <pid> [options]");
+            System.exit(1);
+        }
+        String agentPath = args[0];
+        String pid = args[1];
+        String options = args.length > 2 ? args[2] : "";
+        attachAgent(pid, agentPath, options);
+    }
+
+    static void attachAgent(String pid, String agentPath, String options)
+        throws Exception {
+        VirtualMachine vm = VirtualMachine.attach(pid);
+        try {
+            File lib = new File(agentPath);
+            String fullPath = lib.getAbsolutePath();
+            if (!lib.exists()) {
+                System.out.printf("The agent at %s could not be found",
+                                  fullPath);
+                System.exit(1);
+            }
+            else vm.loadAgentPath(fullPath, options);
+        } finally {
+            vm.detach();
+        }
+    }
+}
```

Comment:
I was actually thinking on including [jattach](https://github.com/apangin/jattach) launcher for this purpose, as it has faster startup and does not require tools.jar available in JDK but not JRE.